engine/src/org/pentaho/di/trans/steps/httppost/HTTPPOST.java

```diff
@@ -235,6 +238,17 @@ public HTTPPOST( StepMeta stepMeta, StepDataInterface stepDataInterface,
                   encoding = contentType.replaceFirst( "^.*;\\s*charset\\s*=\\s*", "" ).replace( "\"", "" ).trim();
                 }
               }
+              
+              headerString.append("{");
```

Comment:
As noted above. Please build JSON using an org.json.simple.JSONObject as shown in JsonOutput step (org.pentaho.di.trans.steps.jsonoutput).